### PR TITLE
[SPARK-7945][CORE]Do trim to values in properties file

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
https://issues.apache.org/jira/browse/SPARK-7945

Now applications submited by org.apache.spark.launcher.Main read properties file without doing trim to values in it.
If user left a space after a value(say spark.driver.extraClassPath) then it probably affect global functions(like some jar could not be included in the classpath), so we should do it like Utils.getPropertiesFromFile.

### CODE REVIEW

**Comment by srowen** [comment]
Yeah, the way `Properties` parses is a little weird, in that whitespace doesn't matter around keys and values except at the end of the value. On the one hand I'm reluctant to make Spark's use of `Properties` nonstandard, but I also can't think of a case where trailing whitespace is valuable to preserve.

This only affects one usage of `Properties` though which seems inconsistent, so I wonder if this is just causing more confusion. It's also not directly addressing the stated problem, which is theoretical, that this is causing JARs to not load.

**Comment by WangTaoTheTonic** [comment]
What more confusion would this cause? And I cann't think of another solution to address this perfectly so did same as `Utils.getPropertiesFromFile`.

**Comment by srowen** [comment]
You expect `Properties` syntax but get something slightly different? A pretty theoretical problem, but that's what I meant.

I suppose we can't use `Utils.getPropertiesFromFile` because is in the launcher, but are these the only places `Properties` are parsed this way? then at least this change would make Spark consistent.

I suppose if `Utils` is already doing this then the cat's out of the bag, so no point in restricting this change to the identified problem I guess.

**Comment by WangTaoTheTonic** [comment]
If you look at this change https://github.com/apache/spark/pull/3916, then will find actually we did use `Utils.getPropertiesFromFile` before. 

I think it is a miss by carelessness. @vanzin Could you help check this either?

**Comment by vanzin** [comment]
Yeah, it's a theoretical problem, but to be consistent with the rest of the code, I guess it doesn't hurt.

File: launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java [hunk 1]
```diff
@@ -295,7 +295,11 @@ Properties loadPropertiesFile() throws IOException {
       FileInputStream fd = null;
       try {
         fd = new FileInputStream(propsFile);
-        props.load(new InputStreamReader(fd, "UTF-8"));
+        Properties rawProps = new Properties();
```
Comment (vanzin) [inline launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java]: This is way more complicated than it needs to be. Just do:

    for (Map.Entry<Object, Object> e : props.entrySet()) {
      e.setValue(e.getValue().toString().trim());
    }

**Comment by WangTaoTheTonic** [comment]
I'm at home now and commit this without compiling locally, hope it all be good.

**Comment by WangTaoTheTonic** [comment]
Jenkins, retest this please.

**Comment by WangTaoTheTonic** [comment]
I've done a test with current code, it worked out fine.

**Comment by srowen** [comment]
LGTM

**Pull request closed without merging**